File: hcx-pipeline-jobs/payments-job/src/main/scala/org/swasth/dp/payments/functions/PaymentsProcessFunction.scala
package org.swasth.dp.payments.functions

import org.apache.flink.api.common.typeinfo.TypeInformation
import org.slf4j.LoggerFactory
import org.swasth.dp.core.function.{BaseDispatcherFunction, ValidationResult}
import org.swasth.dp.payments.task.PaymentsConfig

import java.util

class PaymentsProcessFunction(config: PaymentsConfig)(implicit val stringTypeInfo: TypeInformation[String])
  extends BaseDispatcherFunction(config) {

  private[this] val logger = LoggerFactory.getLogger(classOf[PaymentsProcessFunction])

  override def validate(event: util.Map[String, AnyRef]): ValidationResult = {
    // TODO: Add domain specific validations
    ValidationResult(status = true, None)
  }

}

File: hcx-pipeline-jobs/payments-job/src/main/java/org/swasth/dp/payments/functions/PaymentsProcessFunction.java
package org.swasth.dp.payments.functions;


import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.swasth.dp.core.function.BaseDispatcherFunction;
import org.swasth.dp.core.function.ValidationResult;
import org.swasth.dp.payments.task.PaymentsConfig;

import java.util.Map;

public class PaymentsProcessFunction extends BaseDispatcherFunction {

  private Logger logger = LoggerFactory.getLogger(PaymentsProcessFunction.class);
  private PaymentsConfig config;

  public PaymentsProcessFunction(PaymentsConfig config) {
    super(config);
    this.config = config;
  }

  @Override
  public ValidationResult validate(Map<String, Object> event) {
    // TODO: Add domain specific validations
    return new ValidationResult(true, null);
  }

}